example/MODULE.bazel

```diff
@@ -73,7 +74,10 @@ use_repo(go_sdk, "go_sdk")
 # https://github.com/bazelbuild/rules_jvm_external#pinning-artifacts-and-integration-with-bazels-downloader
 maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
 maven.install(
-    artifacts = ["org.scalameta:scalafmt-cli_2.13:3.7.14"],
+    artifacts = [
+        "org.scalameta:scalafmt-cli_2.13:3.7.14",
+        "com.pinterest.ktlint:ktlint-cli:1.2.1",
```

Comment:
does it ship with a singlejar? The PMD example is worth studying here as a possible alternative to needing a maven lockfile

Reply:
It's not exactly singlejar but it's an executable https://github.com/pinterest/ktlint/blob/master/ktlint-cli/build.gradle.kts#L98 which wraps `java` with the jar file. So I don't think we can use `java_binary` directly but I've updated to using `http_file` and wrapping the actions with Java from the registered Java runtime so it's hermetic.

Reply:
nice, that's much simpler and less likely to cause "too many mavens" or whatever it is rules_jvm_external gets on about